docker/util.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kind/pkg/cluster/constants"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+// clusterLabel returns the label applied to all the containers in a cluster
+func clusterLabel(name string) string {
+	return fmt.Sprintf("%s=%s", constants.ClusterLabelKey, name)
+}
+
+// roleLabel returns the label applied to all the containers with a specific role
+func roleLabel(role string) string {
+	return fmt.Sprintf("%s=%s", constants.NodeRoleKey, role)
+}
+
+// withName returns a filter on name for listContainers & getContainer
+func withName(name string) string {
+	return fmt.Sprintf("name=%s", name)
+}
+
+// withLabel returns a filter on labels for listContainers & getContainer
+func withLabel(label string) string {
+	return fmt.Sprintf("label=%s", label)
+}
+
+// listContainers returns the list of docker containers matching filters
+func listContainers(filters ...string) ([]nodes.Node, error) {
```

Comment:
it appears that this function could be removed and replaced with a direct call to `nodes.List`. Unless there is some reason we want this call wrapped up?

Reply:
I prefer to keep this / ideally, in the long term we can get rid of the kind dependency

Reply:
I agree about removing the kind dependency, but calls to `node` are littered throughout which is why I asked.

But it's fine, let's keep it and we can worry about refactoring later.